src/script/azure_helpers.rb

```diff
@@ -74,16 +76,16 @@ def pull_request_auto_complete(pull_request_id, auto_complete_user_id, merge_str
                 response = patch(source.api_endpoint +
                     source.organization + "/" + source.project +
                     "/_apis/git/repositories/" + source.unscoped_repo +
-                    "/pullrequests/#{pull_request_id}?api-version=5.0", content.to_json)
+                    "/pullrequests/#{pull_request_id}?api-version=6.0", content.to_json)
```

Comment:
Retain version 5.0 for some deployments of Azure DevOps Server

Reply:
5.0 is Azure DevOps Server 2019. 
But with 6.0 the `autoCompleteIgnoreConfigIds` Option is available
see [ApiVersion 6.0](https://learn.microsoft.com/en-us/rest/api/azure/devops/git/pull-requests/create?view=azure-devops-rest-6.0&tabs=HTTP#gitpullrequestcompletionoptions) vs [ApiVersion 5.0](https://learn.microsoft.com/en-us/rest/api/azure/devops/git/pull-requests/create?view=azure-devops-rest-5.0&tabs=HTTP#gitpullrequestcompletionoptions)

Btw. the original implementation without my changes, references to `mergeStrategy` which is not available in 5.0, so it was not working, as in old API this was a boolean `squashMerge: true|false`

And the original implementation already references the 6.0 API [here](https://github.com/tinglesoftware/dependabot-azure-devops/blob/16b52b4819a2ba346744e54a5b30e5eb901d7b45/src/script/azure_helpers.rb#L97) and 6.0-preview.3 [here](https://github.com/tinglesoftware/dependabot-azure-devops/blob/16b52b4819a2ba346744e54a5b30e5eb901d7b45/src/script/azure_helpers.rb#L84)

Reply:
Thanks for these insights. I'm sure they can be useful in https://github.com/dependabot/dependabot-core/pull/4064

Maybe in the future the DevOps REST API version should be configurable? Yet again, it may not be possible to suit all API versions and server installations.